Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -435,20 +431,26 @@ void Interpreter::getStackTrace(JSCell* owner, Vector<StackFrame>& results, size
     if (!callFrame || !maxStackSize)
         return;
 
-    size_t framesCount = 0;
-    size_t maxFramesCountNeeded = maxStackSize + framesToSkip;
-    StackVisitor::visit(callFrame, vm, [&] (StackVisitor&) -> IterationStatus {
-        if (++framesCount < maxFramesCountNeeded)
+    size_t skippedFrames = 0;
+    size_t visitedFrames = 0;
+    StackVisitor::visit(callFrame, vm, [&] (StackVisitor& visitor) -> IterationStatus {
+        if (++skippedFrames <= framesToSkip)
+            return IterationStatus::Continue;
+
+        if (auto* codeBlock = visitor->codeBlock()) {
+            if (codeBlock->ownerExecutable()->implementationVisibility() != ImplementationVisibility::Public)
+                return IterationStatus::Continue;
+        }
+
+        if (++visitedFrames <= maxStackSize)
```

Comment:
How about,
```
if (++visitedFrames < maxStackSize)
```
And just use `visitedFrames` instead of `std::min(visitedFrames, maxStackSize)`?

Reply:
oh lol good call